kythe/go/localrun/localrun.go

```diff
@@ -0,0 +1,518 @@
+/*
+ * Copyright 2020 The Kythe Authors. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+// Package localrun provides helper methods for locall building a kythe indexed
+// repo
+package localrun //import "kythe.io/kythe/go/localrun"
+
+import (
+	"bytes"
+	"context"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"log"
+	"os"
+	"os/exec"
+	"strings"
+	"time"
+
+	"kythe.io/kythe/go/platform/delimited"
+	"kythe.io/kythe/go/platform/delimited/dedup"
+	"kythe.io/kythe/go/services/graphstore"
+	"kythe.io/kythe/go/serving/pipeline"
+	"kythe.io/kythe/go/serving/pipeline/beamio"
+	"kythe.io/kythe/go/serving/xrefs"
+	"kythe.io/kythe/go/util/datasize"
+
+	"github.com/apache/beam/sdks/go/pkg/beam"
+	"github.com/apache/beam/sdks/go/pkg/beam/x/beamx"
+
+	bespb "kythe.io/third_party/bazel/build_event_stream_go_proto"
+)
+
+var _ = delimited.Reader{}
+var _ = dedup.Reader{}
+
+// This list must be kept in sync with Language.String.
+const (
+	Cxx Language = iota // Note that Cxx must be first, or you must update Language.Valid to refer to the first element.
```

Comment:
We can add a dummy value before and after the actual languages to make this clearer:

```
const (
  lowLanguageMarker Language = iota

  Cxx
  Go
  ...

  highLanguageMarker
)

// Valid determines of the provided language is a valid enum value.
func (l Language) Valid() bool {
	return l > lowLanguageMaker && l < highLanguageMarker
}
```

Reply:
Done (both). Good suggestion!